### PR TITLE
Rename DestinationLabel on BottomNavigationBar

### PR DESCRIPTION
**Pull request by lukef**
- Renamed `DestinationLabel` to `BottomNavigationBarItem`
- Renamed `BottomNavigationBarItem.labels` to `BottomNavigationBarItem.items`
- Adjusted comments

### CODE REVIEW

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 1]
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
...
 }
 
 /// A material widget displayed at the bottom of an app for selecting among a
 /// small number of views.
 ///
-/// The bottom navigation bar consists of multiple destinations in the form of
-/// labels laid out on top of a piece of material. It provies quick navigation
-/// between top-level views of an app and is typically used on mobile. For
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple item in the form of
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart]: grammar

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 2]
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
...
+/// labels and/or icons laid out on top of a piece of material. It provides
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart]: prefer "labels, icons, or both"

**Comment by lukef** [comment]
I adjusted the comments based on your feedback. I removed the "primarily used on mobile" as it seems to have been pulled from the material spec directly and its the only real use case right now. I left in a note about larger format devices/screens (it's also in the spec) as it could be useful design advice. Let me know if you want that pulled.

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 3]
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
...
 /// A material widget displayed at the bottom of an app for selecting among a
 /// small number of views.
 ///
-/// The bottom navigation bar consists of multiple destinations in the form of
-/// labels laid out on top of a piece of material. It provies quick navigation
-/// between top-level views of an app and is typically used on mobile. For
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple items in the form of
+/// labels, icons, or both laid out on top of a piece of material. It provides
+/// quick navigation between the top-level views of an app. Note: for larger screens,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart]: Please avoid the word "Note".

See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 4]
```diff
@@ -57,91 +57,91 @@ class DestinationLabel {
...
 
 /// A material widget displayed at the bottom of an app for selecting among a
 /// small number of views.
 ///
-/// The bottom navigation bar consists of multiple destinations in the form of
-/// labels laid out on top of a piece of material. It provies quick navigation
-/// between top-level views of an app and is typically used on mobile. For
-/// larger screens, side navigation may be a better fit.
+/// The bottom navigation bar consists of multiple items in the form of
+/// labels, icons, or both laid out on top of a piece of material. It provides
```
Comment (Hixie) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart]: grammar nit: i'd put a comma after "both" to make it clear that the next part doesn't only apply to the "both" option.

**Comment by Hixie** [comment]
LGTM

**Comment by lukef** [comment]
Did you want me to change the text/grammar? I'll be doing another commit later for a few other things so I could incorporate them later.

**Comment by Hixie** [comment]
Yes for the changes, but it's fine if you do them in the coming days instead of right away.

**Comment by lukef** [comment]
Done

**Comment by Hixie** [comment]
+1 thanks!

**Comment by sethladd** [comment]
Thanks @lukef !

@Hixie is your LGTM sufficient to merge this?

**Comment by sethladd** [comment]
@lukef would you like to add yourself to our AUTHORS file? https://github.com/flutter/flutter/blob/master/AUTHORS

**Comment by lukef** [comment]
Added myself. Thanks! 👍

**Comment by Hixie** [comment]
Merged. FYI @abarth for the API change. We should make sure we've not left any DestinationLabels anywhere.

Luke, would you mind e-mailing flutter-dev to let them know about the API change? Thanks!

**Comment by Hixie** [comment]
And thanks for the contribution! Much appreciated!

**Pull request merged**